fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanDisbursementService.java

```diff
@@ -377,4 +420,64 @@ private Map<String, Object> parseDisbursementDetails(final JsonObject jsonObject
         return returnObject;
     }
 
+    private boolean hasMultipleOrPreDefinedDisbursementDetails(final Loan loan,
+            final Collection<LoanDisbursementDetails> undisbursedDetails) {
+        Collection<LoanDisbursementDetails> allDisbursementDetails = loan.getDisbursementDetails();
+
+        if (undisbursedDetails.size() > 1) {
+            return true;
+        }
+
+        if (allDisbursementDetails.size() > 1 && !undisbursedDetails.isEmpty()) {
+            return true;
+        }
+
+        if (undisbursedDetails.size() == 1) {
+            LoanDisbursementDetails singleDetail = undisbursedDetails.iterator().next();
+            BigDecimal loanPrincipal = loan.getLoanRepaymentScheduleDetail().getPrincipal().getAmount();
+
+            if (singleDetail.principal().compareTo(loanPrincipal) == 0) {
+                return false;
+            }
+        }
+
+        // Default to tranche behavior for safety in ambiguous cases
+        return true;
+    }
+
+    public static List<LoanDisbursementDetails> sortDisbursementDetailsByBusinessRules(
+            Collection<LoanDisbursementDetails> disbursementDetails) {
+        if (disbursementDetails == null || disbursementDetails.isEmpty()) {
+            return List.of();
+        }
+
+        return disbursementDetails.stream()
+                .sorted(Comparator.comparing(LoanDisbursementDetails::expectedDisbursementDate)
+                        .thenComparing((LoanDisbursementDetails d1, LoanDisbursementDetails d2) -> d2.principal().compareTo(d1.principal()))
```

Comment:
I dont think we should order by amount. We should order by the expected disbursement date and created date?